packages/flutter_goldens/lib/src/flutter_goldens_web.dart

```diff
@@ -0,0 +1,107 @@
+/// history of generated goldens over time. Each unique golden gets a unique
+/// color. If the color is the same for all commits in the recent history, the
+/// golden is likely no longer flaky and [isFlaky] can be set back to false. If
+/// the color changes from commit to commit then it is still flaky.
+AsyncMatcher matchesFlutterGolden(Object key, { int? version, bool isFlaky = false }) {
+  if (key is Uri) {
+    return MatchesGoldenFile(key, version, comparator: isFlaky ? _flakyGoldenFileComparator : null);
+  } else if (key is String) {
+    return MatchesGoldenFile.forStringPath(key, version, comparator: isFlaky ? _flakyGoldenFileComparator : null);
+  }
+  throw ArgumentError('Unexpected type for golden file: ${key.runtimeType}');
+}
+
+GoldenComparatorType? _flakyGoldenFileComparator;
+
+/// Main method that can be used in a `flutter_test_config.dart` file to set
+/// [goldenFileComparator] to an instance of [FlutterGoldenFileComparator] that
+/// works for the current test. _Which_ FlutterGoldenFileComparator is
+/// instantiated is based on the current testing environment.
+///
+/// When set, the `namePrefix` is prepended to the names of all gold images.
+Future<void> testExecutable(FutureOr<void> Function() testMain, {String? namePrefix}) async {
+  // The `as GoldenComparatorType` is needed because in the presence of
+  // configurable imports the analyzer picks the io version, not the web
+  // version, and sees `WebGoldenComparator` as incompatible with `GoldenComparatorType`.
+  _flakyGoldenFileComparator = FlakyWebGoldenComparator(webTestUri) as GoldenComparatorType;
+  await testMain();
+}
+
+/// See the io implementation of this function.
+Future<void> processBrowserCommand(dynamic command) async {
+  throw UnimplementedError('processCommand is not used inside the browser');
+}
+
+/// Same as [DefaultWebGoldenComparator] but supports flaky golden checks.
+class FlakyWebGoldenComparator extends WebGoldenComparator {
+  /// Creates a new [FlakyWebGoldenComparator] for the specified [testUri].
+  ///
+  /// Golden file keys will be interpreted as file paths relative to the
+  /// directory in which [testUri] resides.
+  ///
+  /// The [testUri] URL must represent a file.
+  FlakyWebGoldenComparator(this.testUri);
+
+  /// The test file currently being executed.
+  ///
+  /// Golden file keys will be interpreted as file paths relative to the
+  /// directory in which this file resides.
+  Uri testUri;
+
+  @override
+  Future<bool> compare(double width, double height, Uri golden) async {
+    final String key = golden.toString();
+    final html.HttpRequest request = await html.HttpRequest.request(
+      'flutter_goldens',
+      method: 'POST',
+      sendData: json.encode(<String, Object>{
+        'testUri': testUri.toString(),
+        'key': key,
+        'width': width.round(),
+        'height': height.round(),
+        'customProperties': <String, dynamic>{
```

Comment:
Then you may not need these customProperties, if it is just a method rather than a unique comparator that handles the flaky case

Reply:
Hmm, the way web goldens work is the comparator running in the browser isn't comparing anything. It just [sends a command](https://github.com/flutter/flutter/blob/23401885bf3e5abf102d7930fafb8d5a0c96a7cf/packages/flutter_test/lib/src/_goldens_web.dart#L59) to the Flutter Tool and the tool forwards it to a [comparator process](https://github.com/flutter/flutter/blob/23401885bf3e5abf102d7930fafb8d5a0c96a7cf/packages/flutter_tools/lib/src/test/flutter_web_goldens.dart#L122). The comparator process needs to know whether to compare the golden as stable or as flaky. I'm currently using the `customProperties` to attach the `isFlaky` flag. I couldn't think of a different approach.